Gems/Atom/Feature/Common/Code/Source/Debug/RenderDebugFeatureProcessor.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Debug/RenderDebugFeatureProcessor.h>
+#include <Debug/RenderDebugSettings.h>
+
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/Shader/ShaderSystemInterface.h>
+
+namespace AZ {
+    namespace Render {
+
+        RenderDebugFeatureProcessor::RenderDebugFeatureProcessor()
+        {
+        }
+
+        void RenderDebugFeatureProcessor::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                serializeContext
+                    ->Class<RenderDebugFeatureProcessor, RPI::FeatureProcessor>()
+                    ->Version(0);
+            }
+        }
+
+        //! RenderDebugFeatureProcessorInterface overrides...
+        RenderDebugSettingsInterface* RenderDebugFeatureProcessor::GetSettingsInterface()
+        {
+            return m_settings.get();
+        }
+
+        void RenderDebugFeatureProcessor::OnRenderDebugComponentAdded()
+        {
+            ++m_debugComponentCount;
+        }
+
+        void RenderDebugFeatureProcessor::OnRenderDebugComponentRemoved()
+        {
+            --m_debugComponentCount;
+        }
+
+        void RenderDebugFeatureProcessor::Activate()
+        {
+            m_sceneSrg = GetParentScene()->GetShaderResourceGroup();
+            m_settings = AZStd::make_unique<RenderDebugSettings>(this);
+        }
+
+        void RenderDebugFeatureProcessor::Deactivate()
+        {
+            m_sceneSrg = nullptr;
+            m_settings = nullptr;
+        }
+
+        void RenderDebugFeatureProcessor::Simulate(const RPI::FeatureProcessor::SimulatePacket& packet)
+        {
+            AZ_PROFILE_SCOPE(RPI, "RenderDebugFeatureProcessor: Simulate");
+            AZ_UNUSED(packet);
+
+            if (m_settings)
+            {
+                m_settings->Simulate();
+            }
+        }
+
+        void RenderDebugFeatureProcessor::Render(const RPI::FeatureProcessor::RenderPacket& packet)
+        {
+            AZ_PROFILE_SCOPE(RPI, "RenderDebugFeatureProcessor: Render");
+            AZ_UNUSED(packet);
+
+            // Disable debugging if no render debug level component is active
+            bool debugEnabled = (m_debugComponentCount > 0) && m_settings->GetEnabled();
+
+            RPI::ShaderSystemInterface::Get()->SetGlobalShaderOption(m_shaderDebugEnableOptionName, AZ::RPI::ShaderOptionValue{ debugEnabled });
+
+            if (GetParentScene())
+            {
+                Data::Instance<RPI::ShaderResourceGroup> sceneSrg = GetParentScene()->GetShaderResourceGroup();
+
```

Comment:
Don't you already cache this in Activate?

Reply:
Yup, will fix